sys/pool_alloc.c

```diff
@@ -0,0 +1,235 @@
+#include <stdc.h>
+#include <vm.h>
+#include <physmem.h>
+#include <queue.h>
+#include <bitstring.h>
+#include <common.h>
+#include <pool_alloc.h>
+
+#define PI_MAGIC_WORD 0xcafebabe
+
+#define PP_FREED_WORD 0xdeadbeef
+
+#define GET_PI_AT_IDX(slab, i, size)                                           \
+  (pool_item_t *)((unsigned long)slab + (slab->ph_start) +                     \
+                  (i) * (size + sizeof(pool_item_t)))
+
+#define ALIGN_TO_BINARY_POW(x, a) ((((x) >> (a)) + 1) << (a))
+
+#define PALLOC_DEBUG 0
+
+#if defined(PALLOC_DEBUG) && PALLOC_DEBUG > 0
+#define PALLOC_DEBUG_PRINT(text, args...)                                      \
```

Comment:
We don't need yet another macro for displaying messages - just use `log`. I see you wanted to quickly enable/disable pool allocator debug messages and that is indeed useful, but #239 (klog) should be ready very soon and it will enable kernel-wide message filtering. Also, generally, your debug messages are very verbose. Keeping track of the general state of the allocator is great, but displaying messages on every single allocation seems like a bit too much. Most of your messages seem to track control flow ("Entering..."), it it simpler to track it using breakpoints in a debugger. Now that you've tested the allocator and it seems to be working okay, you may want to trim down these messages a bit.